### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by Zilola-Nazarova**
Hi and welcome to this PR 👋🏻 
This branch is following the requirements of [OOP school library: decorate a class](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md) exercise.

### General requirements:
- there are [no linter errors](https://github.com/microverseinc/linters-config)
- correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) is used
- work is documented [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)

### Project requirements:

1. Created `Nameable` class
   - it has `correct_name` method that raises a NotImplementedError
2. Changed `Person` class
   - it inherits from `Nameable`
   - added `correct_name` method that returns the `@name` attribute
3. Created `Decorator`
    - it inherits from `Nameable`
    - it accepts `nameable` object as params and assigns it to `@nameable` instance variable
    - added `correct_name` method that returns the result of the `correct_name` method of the @nameable
4. Created `CapitalizeDecorator` and `TrimmerDecorator`
    - both of them inherit from the base `Decorator` class
    - method `correct_name` in CapitalizeDecorator capitalizes the output of @nameable.correct_name
    - method `correct_name` in TrimmerDecorator trims the word output of @nameable.correct_nameif it is more than 10 characters

5. Tested the decorators:
image

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,8 +1,15 @@
-class Person
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
```
Comment (julie-ify) [inline person.rb]: - [x] **[OPTIONAL]** This class should have its own file.

File: person.rb [hunk 2]
```diff
@@ -13,9 +20,36 @@ def can_use_services?
...
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name[0...10] : @nameable.correct_name
+  end
+end
```
Comment (julie-ify) [inline person.rb]: - [x] **[OPTIONAL]** These 3 classes should have their own files.

**Review by julie-ify — APPROVED** [review]
Hi @Zilola-Nazarova,

## Highlights!
- Good code organization and working decorator ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (Zilola-Nazarova) [reply person.rb]: Thank you @julie-ify for your suggestions!
I made sure to follow them in order to learn Ruby best practices 🙂

**Pull request merged**